great_expectations/core/usage_statistics/anonymizers/batch_request_anonymizer.py

```diff
@@ -47,11 +47,14 @@ def anonymize_batch_request(
             anonymized_batch_request_dict = standardize_batch_request_display_ordering(
                 batch_request=anonymized_batch_request_dict
             )
-            deep_filter_properties_iterable(
-                properties=anonymized_batch_request_dict,
-                clean_falsy=True,
-                inplace=True,
-            )
+
+            # Clean object until all falsy values are removed
```

Comment:
@cdkini Could you please explain why the original single-statement approach does not accomplish the needed goals, while the "while loop" (no pun intended) does?  Perhaps an example or a test (e.g., that endorses the new approach) would help.  Thanks!